### ISSUE TITLE
Ledger is not seen by Electrum wallet

### ISSUE THREAD

**Issue by Talos11** [standard-error/electrum-raven#24]
I can not get Electrum to recognize my ledger.  My computer see's it fine but Electrum just will not.  I am unable to send any coins out.  I really need someone's HELP, Please.
Thanks,
Dan

**Comment by BusterTheFerret** [comment]
Hi Dan,

Have you seen one of these messages?:

"Sign or derivation path is unusual" or "No hardware device detected" ??

If, did you do an firmware update and/or an rvn app update?


Kind Regards,
Dan

**Comment by Talos11** [comment]
No sir, I do not get any error messages.  When  I go into the raven app on the ledger it sounds like it kicks the ledger out of windows, but it doesnt.  It makes the dink donk sound, then when i exit the raven app on the ledger it makes windows i found you sound, dink dink.  I am at my wits end on how I can get my rvn out of this ledger and into a better wallet.  Ledger should make raven a part of the ledger ecosystem, not a red headed step child like it is now with no live support.  
Hope someone can help me.
Dan

**Comment by giotti85** [comment]
> No sir, I do not get any error messages. When I go into the raven app on the ledger it sounds like it kicks the ledger out of windows, but it doesnt. It makes the dink donk sound, then when i exit the raven app on the ledger it makes windows i found you sound, dink dink. I am at my wits end on how I can get my rvn out of this ledger and into a better wallet. Ledger should make raven a part of the ledger ecosystem, not a red headed step child like it is now with no live support.
> Hope someone can help me.
> Dan

I had the same problem. Just close the ledger live software before open electrum. Anyway the electrum wallet still doesn't work with my nano x.

**Comment by Talos11** [comment]
I appreciate the response.  However that does not work for me.  Any time I try to connect to the app on the ledger it disconnects from windows.  Then electrum wallet will not see the ledger. It’s a catch 22. No matter what I do the software will not see the ledger.

**Comment by OrangeFlare** [comment]
I am having the same issues, I can not get my Nano X to setup with the app, I don't even have a wallet yet and the Application cannot recognize my device at all this is consistent across MacOS and Windows I would really like to to store my RVN's Private Key on my Ledger but the Discord has not been helpful and the only app I have been able to get working is wallet.chaintek.net which isn't even that good either

**Comment by aksoftdeveloper** [comment]
Ledger is not getting paired with electrum-raven 3.3.8-rvn3 so not able to send my raven coins to any other address.

**Comment by sobertram** [comment]
Same here. Nano-s.
Ledger firmware 1.6.1
Ravencoin app 1.5.5

**Comment by RainbowMiner** [comment]
Same here
- Leder Nano X
- firmware 1.2.4-5
- Ravencoin App 1.5.5
- Bitcoin App 1.5.5
- Electrum-Raven 3.3.8-rvn3

It worked fine last year, so I suspect one of the past one or two Ledger App updates broke things.

**Comment by RainbowMiner** [comment]
To finally get access to my coins, I have compiled the Electrum-RVN + pull request 24 from source.
This solves the issue. You can download the fixed binaries here:
https://github.com/RainbowMiner/miner-binaries/releases/tag/v3.3.8-rvn3-fixed

Tips are welcome :)
RVN: RYSdLhrkmpjtNGCyboRAs7g7hdujxYyeGc
BTC: 38s4guf2Gu18oS5kjnpt9yAvz37SQCTvBW

**Comment by FibreFoX** [comment]
I can confirm that the version provided from @RainbowMiner it does work 👍

**Comment by Talos11** [comment]
@RainbowMiner   
How does one run the fix??  I have never done anything like what you have done before.  I dont want to lose all my rvn by making a mistake.  How do you do your fix?
Dan

**Comment by FibreFoX** [comment]
@Talos11 look at this comment of a pull-request: https://github.com/standard-error/electrum-raven/pull/23#issuecomment-753588148

You can edit the file yourself while using the original installer, all it needs are 6 additional lines as being available in that pull-request.

**Comment by RainbowMiner** [comment]
@Talos11 @FibreFoX exactly. Just git clone the repository, apply the changes from [pull request 23](https://github.com/standard-error/electrum-raven/pull/23) and compile it. Or go the lazy way and download my compile.

**Comment by stephen304** [comment]
@RainbowMiner Any chance of building an AppImage? I can't seem to build from source without getting missing libraries for every hardware wallet, yet the libraries don't seem to be available via pip.

```
Debug message
  coldcard: (error getting device infos)
    Missing libraries for coldcard.
    Make sure you install it with python3
  keepkey: (error getting device infos)
    Missing libraries for keepkey.
    Make sure you install it with python3
  ledger: (error getting device infos)
    Missing libraries for ledger.
    Make sure you install it with python3
  safe_t: (error getting device infos)
    Missing libraries for safe_t.
    Make sure you install it with python3
  trezor: (error getting device infos)
    Missing libraries for trezor.
    Make sure you install it with python3
```

**Comment by RainbowMiner** [comment]
@stephen304 are you talking about a Ledger Nano S app or the Electrum executable?

**Comment by stephen304** [comment]
@RainbowMiner The python electrum app. The AppImage in the releases here "works" in that it detects my ledger nano x, but fails at the next step. Trying to run from source results in the error I pasted above when it reaches the hardware wallet detection step.

**Comment by Talos11** [comment]
Thanks RainbowMiner, it now see my Ledger nanox but when it asks to confirm it on the ledger I do not get anything to confirm on the ledger.  That is no confirmation comes up on the ledger when signing the transaction.  It just sits there dumb and happy saying application ready.
Any idea what I can do next to try to rescue my coins?

**Comment by BusterTheFerret** [comment]
> 
> 
> Thanks RainbowMiner, it now see my Ledger nanox but when it asks to confirm it on the ledger I do not get anything to confirm on the ledger. That is no confirmation comes up on the ledger when signing the transaction. It just sits there dumb and happy saying application ready.
> Any idea what I can do next to try to rescue my coins?

This is the same with me actually. My wallet runs now but server will not connect and as Talos says its just "aplication ready" and "Not Connected" I can see 1 out of 2 transaction, the first 50RVN as a test but not the second, the interesting one. Server not connected...

KR
